homeassistant/components/zwave/__init__.py

```diff
@@ -1034,6 +1052,16 @@ def value_changed(self):
         self.update_properties()
         self.maybe_schedule_update()
 
+    async def value_renamed(self, update_ids=False):
+        """Rename the node and update any IDs."""
+        self._name = _value_name(self.values.primary)
+        if update_ids:
+            await self.node_removed()
+            self.entity_id = None
```

Comment:
This is not how this works. Nodes have unique IDs and they will get assigned the same entity ID when added back.  You will need to update the entity registry to update the entity ID, the entity instance will pick that up automatically.